build_test.go

```diff
@@ -35,6 +35,11 @@ func TestBuildImageMultipleContextsError(t *testing.T) {
 func TestBuildImageContextDirDockerignoreParsing(t *testing.T) {
 	fakeRT := &FakeRoundTripper{message: "", status: http.StatusOK}
 	client := newTestClient(fakeRT)
+
+	if err := os.Symlink("doesnotexist", "testing/data/symlink"); err != nil {
+		t.Errorf("error creating symlink on demand: %s", err)
+	}
+
```

Comment:
There are some calls to t.Fatal in this test, so I'd prefer to have a defer just after this if:

```
defer func() {
	if err := os.Remove("testing/data/symlink"); err != nil {
		t.Errorf("error removing symlink on demand: %s", err)
	}
}()
```

Thank you!